modules/misc/qt.nix

```diff
@@ -149,18 +179,37 @@ in {
           description = ''
             Theme package to be used in Qt5/Qt6 applications.
             Auto-detected from {option}`qt.style.name` if possible.
+            See its documentation for available options.
           '';
         };
       };
     };
   };
 
   config = let
+    warnGnomeDeprecation = option: name:
+      lib.warnIf (name == "gnome")
+      "The value `gnome` for option `${option}` is deprecated. Use `adwaita` instead."
+      name;
+
+    platformTheme = if (builtins.isString cfg.platformTheme) then {
+      name = lib.warn
+        "The option `qt.platformTheme` has been renamed to `qt.platformTheme.name`."
+        (warnGnomeDeprecation "qt.platformTheme" cfg.platformTheme);
+      package = null;
+    } else if cfg.platformTheme == null then {
+      name = null;
+      package = null;
+    } else {
+      name =
+        warnGnomeDeprecation "cfg.platformTheme.name" cfg.platformTheme.name;
+      package = cfg.platformTheme.package;
+    };
```

Comment:
Could the warnings here be expressed through the `warnings` option instead? Then it would be displayed consistently with other warnings and also be testable. See, e.g., https://github.com/nix-community/home-manager/blob/1c2acec99933f9835cc7ad47e35303de92d923a4/modules/programs/aerc.nix#L172-L182 and https://github.com/nix-community/home-manager/blob/1c2acec99933f9835cc7ad47e35303de92d923a4/tests/modules/programs/aerc/assertion.nix#L12-L20